packages/wallet/backend/src/rapyd/rapyd-client.ts

```diff
@@ -61,86 +102,95 @@ type CalculateSignatureParams = {
 export class RapydClient implements IRapydClient {
   constructor(private deps: RapydClientDependencies) {}
 
-  public createWallet(
+  public async createWallet(
     wallet: RapydWallet
   ): Promise<RapydResponse<RapydWallet>> {
-    return this.post<RapydResponse<RapydWallet>>('user', JSON.stringify(wallet))
+    return this.post('user', JSON.stringify(wallet), RapydWalletSchema)
   }
 
-  public updateProfile(
+  public async updateProfile(
     profile: RapydProfile
   ): Promise<RapydResponse<RapydWallet>> {
-    return this.put<RapydResponse<RapydWallet>>('user', JSON.stringify(profile))
+    return this.put('user', JSON.stringify(profile), RapydWalletSchema)
   }
 
-  public verifyIdentity(
+  public async verifyIdentity(
     req: RapydIdentityRequest
   ): Promise<RapydResponse<RapydIdentityResponse>> {
-    return this.post<RapydResponse<RapydIdentityResponse>>(
+    return this.post(
       'identities',
-      JSON.stringify(req)
+      JSON.stringify(req),
+      RapydIdentityResponseSchema
     )
   }
 
-  public depositLiquidity(
+  public async depositLiquidity(
     req: RapydDepositRequest
   ): Promise<RapydResponse<RapydDepositResponse>> {
-    return this.post<RapydResponse<RapydDepositResponse>>(
+    return this.post(
       'account/deposit',
-      JSON.stringify(req)
+      JSON.stringify(req),
+      RapydDepositResponseSchema
     )
   }
 
-  public holdLiquidity(
+  public async holdLiquidity(
     req: RapydHoldRequest
   ): Promise<RapydResponse<RapydHoldResponse>> {
-    return this.post<RapydResponse<RapydHoldResponse>>(
+    return this.post(
       'account/balance/hold',
-      JSON.stringify(req)
+      JSON.stringify(req),
+      RapydHoldResponseSchema
     )
   }
 
-  public releaseLiquidity(
+  public async releaseLiquidity(
     req: RapydReleaseRequest
   ): Promise<RapydResponse<RapydReleaseResponse>> {
-    return this.post<RapydResponse<RapydReleaseResponse>>(
+    return this.post(
       'account/balance/release',
-      JSON.stringify(req)
+      JSON.stringify(req),
+      RapydReleaseResponseSchema
     )
   }
 
-  public getDocumentTypes(
+  public async getDocumentTypes(
     country: string
   ): Promise<RapydResponse<RapydDocumentType[]>> {
-    return this.get<RapydResponse<RapydDocumentType[]>>(
+    return this.get(
       `identities/types?country=${country}`
-    )
+      // RapydDocumentsTypeSchema
```

Comment:
I am having trouble in validating and infering types in cases where we are expecting arrays as a result from rapyd.

For now, all these cases just don't have validation applied, and I made the private request method also be able to just blindly return responses without any validation or type inference.

Any suggestions or contributions here are welcome

Reply:
maybe make every schema to have the `data` and `status` field that rapyd returns so every response will be an object

Reply:
So you suggest I give up the following structure that I am currently using, in which RapydResponse has a data object of type *whatever rapyd response type*:

`export interface RapydResponse<T> {
  status: z.TypeOf<typeof StatusSchema>
  data: T
}`

used like:

`RapydResponse<RapydDepositResponse>`


and just add the status + data items in every rapyd response type / schema ?

Reply:
yes, as an workaround to not deal with array response type

Reply:
@raducristianpopa  @rico191013  any other thoughts here? I'm still hoping to get the types working with the current structure which implies the generics used in RapydResponse:
`export interface RapydResponse<T> { status: z.TypeOf<typeof StatusSchema> data: T }`

If not, I will prolly go for what Dragos suggested.